tests/testProfileChangePassword.php

```diff
@@ -81,58 +81,135 @@ public function testInitHooks() {
+		$this->assertTrue( $change_password->validate_new_password( $errors, false ) );
+		$this->assertEmpty( $errors->get_error_messages() );
+
+		// Test core WP profile update screen field.
+		unset( $_POST['user_id'] );
 		$this->assertFalse( $change_password->validate_new_password( $errors, false ) );
 
-		$this->storeAuth0Data( $user_id, 'not-auth0' );
+		// Test user object.
+		$this->assertTrue( $change_password->validate_new_password( $errors, $user_obj ) );
+	}
 
-		// Call should fail with Auth0 data stored because of a wrong strategy.
-		$this->assertFalse( $change_password->validate_new_password( $errors, false ) );
+	/**
+	 * Test that a user without Auth0 data or with a non-DB strategy skips update.
+	 */
+	public function testThatNonAuth0UserSkipsUpdate() {
+		$user_id = $this->createUser()->ID;
+		$errors  = new WP_Error();
+
+		$mock_api_test_password = $this->getStub( true );
+		$change_password        = new WP_Auth0_Profile_Change_Password( $mock_api_test_password );
 
+		$_POST['pass1']   = uniqid();
+		$_POST['user_id'] = $user_id;
 		$this->storeAuth0Data( $user_id );
+		$this->assertTrue( $change_password->validate_new_password( $errors, false ) );
+		$this->assertEmpty( $errors->get_error_messages() );
+
+		// Test that an unlinked user will not be updated.
+		self::$users_repo->delete_auth0_object( $user_id );
+		$this->assertFalse( $change_password->validate_new_password( $errors, false ) );
+
+		// Test that a linked, non-DB user will not be updated.
+		$this->storeAuth0Data( $user_id, 'not-a-db-strategy' );
+		$this->assertFalse( $change_password->validate_new_password( $errors, false ) );
+	}
 
-		// Call should succeed with a mocked API.
+	/**
+	 * Test that an API failure will set UI banners and cancel the password change.
+	 */
+	public function testThatApiFailureSetsErrorsUnsetsPassword() {
+		$user_id = $this->createUser()->ID;
+		$errors  = new WP_Error();
+
+		// API call mocked to succeed.
+		$mock_api_test_password = $this->getStub( true );
+		$change_password        = new WP_Auth0_Profile_Change_Password( $mock_api_test_password );
+
+		// Confirm that data is set to succeed.
+		$_POST['pass1']   = uniqid();
+		$_POST['pass2']   = $_POST['pass1'];
+		$_POST['user_id'] = $user_id;
+		$this->storeAuth0Data( $user_id );
 		$this->assertTrue( $change_password->validate_new_password( $errors, false ) );
+		$this->assertEmpty( $errors->get_error_messages() );
+
+		// API call mocked to fail.
+		$mock_api_test_password = $this->getStub( false );
+		$change_password        = new WP_Auth0_Profile_Change_Password( $mock_api_test_password );
 
-		// Call should fail on the second call with a mocked API.
 		$this->assertFalse( $change_password->validate_new_password( $errors, false ) );
 		$this->assertEquals( 'Password could not be updated.', $errors->errors['auth0_password'][0] );
 		$this->assertEquals( 'pass1', $errors->error_data['auth0_password']['form-field'] );
 		$this->assertFalse( isset( $_POST['pass1'] ) );
```

Comment:
why would this fields get unset after a failure?

Reply:
Documented in the method itself but it halts the password change from happening when coming from core.